runtime/vm/gphandle.c

```diff
@@ -347,7 +347,14 @@ throwConditionException(J9VMThread *vmThread, struct ConditionExceptionConstruct
 
 	cls = (*env)->FindClass(env, "com/ibm/le/conditionhandling/ConditionException");
 	if (NULL == cls) {
-		goto cleanup;
+		/* Since this class belongs to a module other than java.base, load another type of
+		 * exception to provide more information about what went wrong.
+		 */
+		cls = (*env)->FindClass(env, "java/lang/ClassNotFoundException");
```

Comment:
I'm not convinced using CNFE is a good alternative to ConditionException. Code that expects to catch ConditionException will fail as CNFE won't match the `catch` clause.

Also, the `cls` is used to lookup an `<init>` method with the signature `"(Ljava/lang/String;J[BLjava/lang/String;IIIIIJ)V"` which I'd be surprised to find on CNFE.

Reply:
We may need to find a way to have the `openj9.zosconditionhandling` register this class with the VM so we can use it from native code.

Reply:
CNFE seems reasonable when ConditionException is not available, as the problem would be that the openj9.zosconditionhandling module is missing from the runtime. However it should include a message like "com.ibm.le.conditionhandling.ConditionException not found, include the openj9.zosconditionhandling module". 

I don't think code that catches ConditionException would even be able to load in the absence of ConditionException. I tried a simple testcase and it throws a ClassNotFoundException when a caught exception is missing, although the exception was never thrown.

```
public class TestMissingException {
	
	public static void test() throws Throwable {
	}

	public static void main(String[] args) throws Throwable {
		try {
			test();
		} catch (TestException e) {
}
}
```

Reply:
I don't have a problem with throwing a CNFE if its due to not finding the `ConditionException` class.  If that's the case, the original `FindClass()` JNI call should leave a pending exception along with its null return value.

Bottom line: We don't need to explicitly write this code as there will already be a pending exception.

Reply:
okay. I'll remove it.